File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/implement/AccountConvertImp.java
package com.platdmit.simplecloudmanager.domain.converters.implement;

import com.platdmit.simplecloudmanager.data.api.models.ApiAccount;
import com.platdmit.simplecloudmanager.data.database.entity.DbAccount;
import com.platdmit.simplecloudmanager.domain.converters.AccountConverter;
import com.platdmit.simplecloudmanager.domain.models.UserAccount;

public class AccountConvertImp implements AccountConverter {
    @Override
    public DbAccount fromApiToDb(ApiAccount apiAccount, String password) {
        return new DbAccount(
                apiAccount.getUuid(),
                apiAccount.isSubaccount(),
                apiAccount.getLogin(),
                apiAccount.getEmail(),
                password,
                apiAccount.getBalance(),
                apiAccount.getVpsLimit()
        );
    }

    @Override
    public UserAccount fromApiToDomain(ApiAccount apiAccount, String password) {
        return new UserAccount(
                apiAccount.getUuid(),
                true,
                apiAccount.isSubaccount(),
                apiAccount.getLogin(),
                password,
                apiAccount.getBalance(),
                apiAccount.getVpsLimit(),
                ""
        );
    }

    @Override
    public UserAccount fromDbToDomain(DbAccount dbAccount) {
        return new UserAccount(
                dbAccount.getId(),
                dbAccount.isMain(),
                dbAccount.isSubAccount(),
                dbAccount.getLogin(),
                dbAccount.getPass(),
                dbAccount.getBalance(),
                dbAccount.getVpsLimit(),
                dbAccount.getPin()
        );
    }

    @Override
    public DbAccount fromDomainToDb(UserAccount userAccount) {
        return new DbAccount(
                userAccount.getId(),
                userAccount.isSubAccount(),
                userAccount.getLogin(),
                userAccount.getEmail(),
                userAccount.getPass(),
                userAccount.getBalance(),
                userAccount.getVpsLimit()
        );
    }
}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/implement/AccountConvertImp.kt
package com.platdmit.simplecloudmanager.domain.converters.implement

import com.platdmit.data.api.models.ApiAccount
import com.platdmit.data.database.entity.DbAccount
import com.platdmit.simplecloudmanager.domain.converters.AccountConverter
import com.platdmit.simplecloudmanager.domain.models.UserAccount

class AccountConvertImp : AccountConverter {
    override fun fromApiToDb(apiAccount: ApiAccount, password: String): DbAccount {
        return DbAccount(
                apiAccount.uuid.toLong(),
                apiAccount.isSubaccount,
                apiAccount.login,
                apiAccount.email,
                password,
                apiAccount.balance,
                apiAccount.vpsLimit
        )
    }

    override fun fromApiToDomain(apiAccount: ApiAccount, password: String): UserAccount {
        return UserAccount(
                apiAccount.uuid.toLong(),
                apiAccount.isSubaccount,
                apiAccount.login,
                password,
                apiAccount.balance,
                apiAccount.vpsLimit
        )
    }

    override fun fromDbToDomain(dbAccount: DbAccount): UserAccount {
        return UserAccount(
                dbAccount.id,
                dbAccount.isSubAccount,
                dbAccount.login,
                dbAccount.pass,
                dbAccount.balance,
                dbAccount.vpsLimit,
                dbAccount.isMain,
                dbAccount.pin.orEmpty()
        )
    }

    override fun fromDomainToDb(userAccount: UserAccount): DbAccount {
        return DbAccount(
                userAccount.id,
                userAccount.subAccount,
                userAccount.login,
                userAccount.email,
                userAccount.pass,
                userAccount.balance,
                userAccount.vpsLimit
        )
    }
}